odoo/addons/base/models/ir_model.py

```diff
@@ -156,6 +160,22 @@ def _check_model_name(self):
             if not models.check_object_name(model.model):
                 raise ValidationError(_("The model name can only contain lowercase characters, digits, underscores and dots."))
 
+    @api.constrains('default_order')
+    def _check_order(self):
+        for model in self:
+            if self.env.get(model.model) is None:
+                # magic fields (like id) are not set up yet and registry
+                # is not updated, can't validate anything
+                # field is invisible in the view until first save + default is 'id'
+                # should be sufficient to assume things will go well
+                continue
+            stored_fields = list(map(lambda f: f.name, filter(lambda f: f.store == True, self.env[model.model]._fields.values())))
+            model._check_qorder(model.default_order)  # regex check for the whole clause ('is it valid sql?')
+            order_fields = re.findall(r'(?:\s*\"?([a-z0-9:_]+)\"?(?:\s+(?:desc|asc))?\s*(?:,|$))', model.default_order)
```

Comment:
The following regexp might be simpler: `r'"?(\w+)"?\s*(?:asc|desc)?'`and should be matched with `re.I`.